src/EnergyPlus/WeatherManager.hh

```diff
@@ -774,95 +774,98 @@ struct WeatherManagerData : BaseGlobalStruct
-    Array1D_int WeekDayTypes;    // To hold Week day types using specified first day NOLINT(cert-err58-cpp)
-    Array1D_int DSTIndex;        // To hold DST Index based on weather file or input NOLINT(cert-err58-cpp)
-
-    int NumDataPeriods;
-
-    int NumIntervalsPerHour;
-
-    bool UseDaylightSaving;         // True if user says to use Weather File specified DaylightSaving Period
-    bool UseSpecialDays;            // True if user says to use Weather File specified Special Days for current RunPeriod
-    bool UseRainValues;             // True if rain values from weather file are to be used
-    bool UseSnowValues;             // True if snow values from weather file are to be used
-    bool EPWDaylightSaving;         // True if a DaylightSaving Time Period is input (EPW files)
-    bool IDFDaylightSaving;         // True if a DaylightSaving Time Period is input (IDF files)
-    bool DaylightSavingIsActive;    // True if a DaylightSavingPeriod should be used for Environment
-    bool WFAllowsLeapYears;         // True if the Weather File (WF) header has "Yes" for Leap Years
-    int curSimDayForEndOfRunPeriod; // normal=number days in sim, but different when repeating runperiods or multi-year files
-    int Envrn;                      // Counter for environments
-    int NumOfEnvrn;                 // Number of environments to be simulated
-    int NumEPWTypExtSets;           // Number of Typical/Extreme on weather file.
-    int NumWPSkyTemperatures;       // Number of WeatherProperty:SkyTemperature items in input file
+    bool GetBranchInputOneTimeFlag = true;
+    bool GetEnvironmentFirstCall = true;
+    bool PrntEnvHeaders = true;
+    bool FirstCall = true;                 // Some things should only be done once
+    bool WaterMainsParameterReport = true; // should only be done once
+    bool PrintEnvrnStamp = false;          // Set to true when the environment header should be printed
+    bool PrintDDHeader = true;
+
+    int YearOfSim = 1;                     // The Present year of Simulation.
+    int NumDaysInYear = 365;               // TODO: removed const from this until leap year behavior is reviewed
+    int EnvironmentReportNbr = 0;          // Report number for the environment stamp
+    std::string EnvironmentReportChr = ""; // Report number for the environment stamp (character -- for printing)
+    bool WeatherFileExists = false;        // Set to true if a weather file exists
+    std::string LocationTitle = "";        // Location Title from input File
+    bool LocationGathered = false;         // flag to show if Location exists on Input File (we assume one is there and correct on weather file)
+
+    Real64 WeatherFileLatitude = 0.0;
+    Real64 WeatherFileLongitude = 0.0;
+    Real64 WeatherFileTimeZone = 0.0;
+    Real64 WeatherFileElevation = 0.0;
+    Array1D<Real64> GroundTempsFCFromEPWHeader = Array1D<Real64>(12, 0.0); // F or C factor method NOLINT(cert-err58-cpp)
```

Comment:
These known length Array1D should be made into std::array ASAP, but not necessarily here.